### PR TITLE
Add data manupulation to model

### PR DESCRIPTION
**Pull request by aimemalaika**
## What does this PR do?
This PR is raised to add methods to manipulate data in models
## Description of the task to be completed
- Add update_post_counter in Post
- Add recent_comments in Post
- Add update_comments_counter in Comments
- Add update_likes_counter in  Likes
- Add recent_posts in User
## How should it be tested
- clone the repository\
  `git clone https://github.com/aimemalaika/blog.git`
- navigate to the folder\
  `cd blog`
- checkout the feature\
  ``- navigate to the folder\
    `cd blog`
- Install packages\
    `bundle install`
- Setup database\
    `rails db:setup`
- Run migration\
    `rails db:migrate`

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,9 @@
 class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
+  has_many :comments, foreign_key: :author_id
+
+  def recent_posts
+    posts.order(created_at: :desc).limit(5)
```
Comment (EricMbouwe) [inline app/models/user.rb]: - [ ] This method should return 3 most recent posts not 5. Let's fix that

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
### Hi @aimemalaika,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlight!
- Good PR ✔️
- Good commits ✔️

# Required Changes ♻️ ⛔️
- [ ] ApplicationController should contain #current_user method. Let's implement it

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb [hunk 2]
```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :likes, foreign_key: :post_id
+  has_many :comments, foreign_key: :post_id
+
+  def update_post_counter
+    author.increment!(:posts_counter)
+  end
+
```
Comment (MahmoudBakr23) [inline app/models/post.rb:11]: **OPTIONAL**
- It'd be nice if you created a custom method using the `after_save` rails callback to call the `update_post_counter` method every time the model gets updated!

Ex:
`after_save :update_post_counter`

> Please consider this suggestion in the other models

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @aimemalaika 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done addressing the previous changes

**Optional**
_Please see the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**